src/main/java/edu/cornell/kfs/fp/batch/service/impl/AwsAccountingXmlDocumentAccountingLineServiceImpl.java

```diff
@@ -41,222 +43,268 @@
+        validateAndSetDefaultsProjectCode(xmlAccountingLine, defaultAccountDto, logErrorMessage);
+        validateAndSetDefaultsOrgRefId(xmlAccountingLine, defaultAccountDto, logErrorMessage);
+    }
 
-    private AccountingXmlDocumentAccountingLine fixAccountFieldReferences(AccountingXmlDocumentAccountingLine xmlAccountingLine, String defaultAccountString) {
-        if (!validateChartCode(xmlAccountingLine.getChartCode())) {
-            String defaultChartCode = getDefaultChartCodeFromParameter();
-            xmlAccountingLine.setChartCode(defaultChartCode);
+    private void validateAndSetDefaultsChart(AccountingXmlDocumentAccountingLine xmlAccountingLine, AmazonKfsAccountDTO defaultAccountDto,
+            boolean logErrorMessage) {
+        if (!validateChartCode(xmlAccountingLine.getChartCode(), logErrorMessage)) {
+            xmlAccountingLine.setChartCode(defaultAccountDto.getKfsChart());
         }
+    }
 
-        if (StringUtils.equalsIgnoreCase(CuFPConstants.AmazonWebServiceBillingConstants.ACCOUNT_NONE, xmlAccountingLine.getAccountNumber())) {
-            xmlAccountingLine = getDefaultAccountingLine(defaultAccountString);
+    private void validateAndSetDefaultsAccount(AccountingXmlDocumentAccountingLine xmlAccountingLine,
+            AmazonKfsAccountDTO defaultAccountDto, boolean logErrorMessage) {
+        if (StringUtils.equalsIgnoreCase(CuFPConstants.AmazonWebServiceBillingConstants.ACCOUNT_NONE, xmlAccountingLine.getAccountNumber()) 
+                || StringUtils.equalsIgnoreCase(CuFPConstants.AmazonWebServiceBillingConstants.INTERNAL_KFS_ACCOUNT_DESCRIPTION, xmlAccountingLine.getAccountNumber()) 
+                || !validateAccount(xmlAccountingLine.getChartCode(), xmlAccountingLine.getAccountNumber(), logErrorMessage)) {
+            xmlAccountingLine.setChartCode(defaultAccountDto.getKfsChart());
+            xmlAccountingLine.setAccountNumber(defaultAccountDto.getKfsAccount());
         }
+    }
 
-        if (!StringUtils.equalsIgnoreCase(defaultAccountString, CuFPConstants.AmazonWebServiceBillingConstants.INTERNAL_KFS_ACCOUNT_DESCRIPTION) &&
-                !validateAccount(xmlAccountingLine.getChartCode(), xmlAccountingLine.getAccountNumber())) {
-            xmlAccountingLine = getDefaultAccountingLine(defaultAccountString);
+    private void validateAndSetDefaultsSubAccount(AccountingXmlDocumentAccountingLine xmlAccountingLine,
+            AmazonKfsAccountDTO defaultAccountDto, boolean logErrorMessage) {
+        if (StringUtils.isNotBlank(xmlAccountingLine.getSubAccountNumber())
+                && !validateSubAccountNumber(xmlAccountingLine.getChartCode(), xmlAccountingLine.getAccountNumber(), xmlAccountingLine.getSubAccountNumber(), logErrorMessage)) {
+            if (validateSubAccountNumber(xmlAccountingLine.getChartCode(), xmlAccountingLine.getAccountNumber(), defaultAccountDto.getKfsSubAccount(), logErrorMessage)) {
```

Comment:
what happens if default sub Account is invalid?

Reply:
it gets set to an empty string